invoke-ai/InvokeAI
# Psyche/feat/UI/regional prompting

## Summary

Add MVP regional prompt UI. It's only accessible in storybook right now. Run `pnpm storybook` and click `RegionalPrompts` -> `Docs`. 

- [x] Layers can be arranged
- [x] Brush and eraser per layer
- [x] Export individual layer images
- [ ] Resizable canvas (currently locked to 512x512)
- [ ] Transformable layers (e.g. position the layer in the canvas by dragging it, scale it, etc.)
- [ ] Integrate in the linear UI (probably separate PR)
- [ ] Wire it up to the graphs (probably separate PR)

https://github.com/invoke-ai/InvokeAI/assets/4822129/83151da4-302f-496f-8339-e5a3030a41f0

## Related Issues / Discussions



## QA Instructions

Usage in its current state:
- Start the UI and storybook.
- Draw your layers and prompts in storybook. Just do 4 for now.
- Click `DEBUG LAYERS` - this will open a new tab for each layer with an image.
- Copy each image, then paste into the app to upload. You this won't work if you are on the workflow editor tab, you need to be on one of the linear tabs.
- Load the sample workflow: 
[SDXL Arbitrary Regional Prompts.json](https://github.com/invoke-ai/InvokeAI/files/14917474/SDXL.Arbitrary.Regional.Prompts.json)
- Drag your images each of the `Resize Image` nodes.
- Update your prompts (manually). I'm not sure if you should set the "style" prompt for these.
- Update the "style" prompt for the global positive prompt (leave "prompt" empty).
- Leave the global iterations count at one. Adjust the integer range to control how many images are generated.
- Invoke!

![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/4fcb49f1-69b5-44d2-adb0-26576e2241b2)

Example outputs from the above setup:

![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/0d29b117-c2a4-4adb-96ad-763dade6eb72)
![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/ccd14820-e2a3-4274-ac80-2fb8c0c6c2c6)
![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/97af0fad-3fc7-4e8b-90d9-a0847b95621b)
![image](https://github.com/invoke-ai/InvokeAI/assets/4822129/8d0c2824-a872-4241-87fe-fcd3ba7aaa51)


## Merge Plan

This PR is based on #5916, but it should not be merged.

## Checklist
